File: frontend/src/types/market.ts
export interface Ticker {
  symbol: string;
  last: number;
  percentage: number;
  quote_volume: number;
  base_volume: number;
  high: number;
  low: number;
}

export interface Kline {
  symbol: string;
  open_time: string;
  close_time: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  quote_volume: number;
  trade_count: number;
}

export interface Trade {
  symbol: string;
  trade_id: number;
  price: number;
  quantity: number;
  quote_qty: number;
  trade_time: string;
  is_buyer_maker: number;
}

export interface WsTradeMessage {
  type: "trade";
  data: Trade;
}

File: backend/app/schemas/market.py
"""API response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class KlineOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    symbol: str
    open_time: datetime
    close_time: datetime
    open: float
    high: float
    low: float
    close: float
    volume: float
    quote_volume: float
    trade_count: int


class TradeOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    symbol: str
    trade_id: int
    price: float
    quantity: float
    quote_qty: float
    trade_time: datetime
    is_buyer_maker: int


class TickerOut(BaseModel):
    symbol: str
    last: float = Field(description="현재가")
    percentage: float = Field(description="24시간 변동률(%)")
    quote_volume: float = Field(description="24시간 거래대금(USDT)")
    base_volume: float = Field(description="24시간 거래량(코인)")
    high: float
    low: float


class HealthOut(BaseModel):
    status: str
    symbols: list[str]
